app/controllers/hackdays_controller.rb

```diff
@@ -0,0 +1,77 @@
+class HackdaysController < ApplicationController
+
+  def index
+    @hackday = Hackday.new
+    set_current_and_past_hackdays
+  end
+
+  def create
+    @hackday = Hackday.new hackday_params
+    
+    if @hackday.save
+      cookies["#{@hackday.id}_votes"] = MAX_VOTES
+      redirect_to @hackday
+    else
+      set_current_and_past_hackdays
+      render :index
+    end
+  end
+
+  def edit
+    @hackday = Hackday.find(params[:id])
+    unless @hackday.active?
+      flash[:warning] = "Cannot edit a non-active hackday"
+      redirect_to request.referrer || root_url
+    end
+  end
+
+  def update
+    @hackday = Hackday.find(params[:id])
+    if @hackday.update_attributes(hackday_params)
+      flash[:success] = "Hackday updated!"
+      redirect_to @hackday
+    else
+      render :edit
+    end
+  end
+
+  def destroy
+    @hackday = Hackday.find(params[:id])
+    if @hackday.destroy
+      cookies.delete "#{@hackday.id}_votes"
+    else
+      flash[:warning] = "Cannot remove a non-active hackday"
+    end
+    redirect_to hackdays_url
+  end
+
+  def show
+    @hackday = Hackday.find(params[:id])
+    @new_project = @hackday.projects.build
+    @projects = @hackday.projects.where(:hackday_id => @hackday.id)
```

Comment:
I included the `where` because the line above (`@hackday.projects.build`) ended up placing the empty newly created project within `@hackday.projects` and I didn't really know how to get just the projects that were saved in the db